sdk/eventhubs/azure-messaging-eventhubs/README.md

```diff
@@ -1,3 +1,189 @@
+
+<!-- NOTE: GO Links are intentional for now -->
+#### Using a connection string
+ - ConsumerClient: [link](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs#example-NewConsumerClientFromConnectionString)
+ - ProducerClient: [link](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs#example-NewProducerClientFromConnectionString)
+
+# Key concepts
+
+An Event Hub [**namespace**](https://docs.microsoft.com/azure/event-hubs/event-hubs-features#namespace) can have multiple event hubs.
+Each event hub, in turn, contains [**partitions**](https://docs.microsoft.com/azure/event-hubs/event-hubs-features#partitions) which 
+store events.
+
+Events are published to an event hub using an [event publisher](https://docs.microsoft.com/azure/event-hubs/event-hubs-features#event-publishers). In this package, the event publisher is the [ProducerClient](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs#ProducerClient)
+
+Events can be consumed from an event hub using an [event consumer](https://docs.microsoft.com/azure/event-hubs/event-hubs-features#event-consumers). In this package there are two types for consuming events: 
+- The basic event consumer is the PartitionClient, in the
+- [ConsumerClient](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs#ConsumerClient). This consumer is useful if you already known which partitions you want to receive from.
+- A distributed event consumer, which uses Azure Blobs for checkpointing and coordination. This is implemented in the [Processor](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs#Processor). 
+This is useful when you want to have the partition assignment be dynamically chosen, and balanced with other Processor instances.
+
+For more information about Event Hubs features and terminology can be found here: [link](https://docs.microsoft.com/azure/event-hubs/event-hubs-features)
+
+
+# Examples
+
+<!-- NOTE: GO Links are intentional for now -->
+Examples for various scenarios can be found on [pkg.go.dev](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs#pkg-examples) or in the example*_test.go files in our GitHub repo for [azeventhubs](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/messaging/azeventhubs).
+
+## Send events
+
+The following example shows how to send events to an event hub:
+
+```cpp
+// Your Event Hubs namespace connection string is available in the Azure portal.
+std::string connectionString = "<connection_string>";
+std::string eventHubName = "<event_hub_name>";
+
+Azure::Messaging::EventHubs::EventDataBatchOptions batchOptions;
+batchOptions.PartitionId = "1";
+Azure::Messaging::EventHubs::EventDataBatch eventBatch(batchOptions);
+
+Azure::Messaging::EventHubs::Models::EventData message;
+message.Body.Data = {'H', 'e', 'l', 'l', 'o', '2'};
+
+eventBatch.AddMessage(message);
```

Comment:
Just looking at the samples, the naming of types maybe be better represented.

Should we consider renaming `EventData` to be `EventMessage`, given it is passed on to `AddMessage` and the variable is named `message`. Unless that's reserved from something else.

Reply:
There are two kinds of messages consumed by the eventhubs client: `EventData`, which is data specific to eventhubs and `AmqpMessage` which is data specific to AMQP.

The GO and C# SDK call the type specific to EventHubs `EventData`, I'm really not comfortable changing the name.